store/sqlstore/plugin_store.go

```diff
@@ -71,6 +71,43 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) store.StoreChann
 	})
 }
 
+func (ps SqlPluginStore) CompareAndUpdate(kv *model.PluginKeyValue, oldValue []byte) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		if result.Err = kv.IsValid(); result.Err != nil {
+			return
+		}
+
+		if ps.DriverName() == model.DATABASE_DRIVER_POSTGRES || ps.DriverName() == model.DATABASE_DRIVER_MYSQL {
```

Comment:
I /think/ we can just discard the `DriverName` test here -- the only other database we support is SQLite, and that only for tests, so since the code is the same for MySQL and PostgreSQL here, it should work fine on SQLite too.